pkg/fixedpoint/value_map.go

```diff
@@ -0,0 +1,135 @@
+package fixedpoint
+
+type ValueMap map[string]Value
+
+func (m ValueMap) Eq(n ValueMap) bool {
+	if len(m) != len(n) {
+		return false
+	}
+
+	for m_k, m_v := range m {
+		n_v, ok := n[m_k]
+		if !ok {
+			return false
+		}
+
+		if !m_v.Eq(n_v) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (m ValueMap) Add(n ValueMap) ValueMap {
+	if len(m) != len(n) {
+		panic("unequal length")
+	}
+
+	o := ValueMap{}
+
+	for k, v := range m {
+		o[k] = v.Add(n[k])
+	}
+
+	return o
+}
+
+func (m ValueMap) Sub(n ValueMap) ValueMap {
+	if len(m) != len(n) {
+		panic("unequal length")
+	}
+
+	o := ValueMap{}
+
+	for k, v := range m {
+		o[k] = v.Sub(n[k])
+	}
+
+	return o
+}
+
+func (m ValueMap) Mul(n ValueMap) ValueMap {
+	if len(m) != len(n) {
+		panic("unequal length")
+	}
+
+	o := ValueMap{}
+
+	for k, v := range m {
+		o[k] = v.Mul(n[k])
+	}
+
+	return o
+}
+
+func (m ValueMap) Div(n ValueMap) ValueMap {
+	if len(m) != len(n) {
+		panic("unequal length")
+	}
+
+	o := ValueMap{}
+
+	for k, v := range m {
+		o[k] = v.Div(n[k])
+	}
+
+	return o
+}
+
+func (m ValueMap) AddScalar(x Value) ValueMap {
```

Comment:
I recommend using `Add(interface{})` similar to `types.Add` in `types/indicator.go`.
Users could pass in either a `ValueMap` or a `fixedpoint.Value` or even a `float64` to do the job.

Reply:
Ok, maybe we can create another PR to do this.